hedera-node/docs/design/services/smart-contract-service/RedirectContract-description.md

```diff
@@ -0,0 +1,224 @@
+
+The "redirect" contracts (aka proxy contracts) implement the feature where you can call a Hedera
+token or account or scheduled transaction directly at their address and they _act_ as if you made
+the corresponding contract call.
+
+There is one redirect contract which is parameterized as a _template_ with the addresses of the
+HTS, HAS, and HSS system contracts, and with the correct "redirect" method selector.
+
+When a call is made to a token, an account, or a scheduled transaction, the redirect template is
+instantiated with the correct system contract address and selector, and then executed.  It wraps
+the calldata (provided by the calling contract or transaction) and delegate-calls the actual
+system contract, passing back whatever the return status and value is.
+
+Expanding the template is done in
+`DispatchingEvmFrameState.get{Token,Account,Schedule}RedirectCode()`.
+
+## Raw Bytecode
+
+Derived from [`DispatchingEvmFrameState` at `e70582e28e`](https://github.com/hashgraph/hedera-services/blob/f023153207482730f79fd5da69c6fbd29eaa3fcc/hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/store/contracts/HederaEvmWorldStateTokenAccount.java#L35) (as viewed on 2024-12-08)
+
+```
+6080604052348015600f57600080fd5b50600061RRRR905077SSSSSSSSfefefe
+fefefefefefefefefefefefefefefefefe600052366000602037600080366018
+016008845af43d806000803e8160008114605857816000f35b816000fdfea264
+6970667358221220d8378feed472ba49a0005514ef7087017f707b45fb9bf56b
+b81bb93ff19a238b64736f6c634300080b0033
+```
+
+Where:
+* `RRRR` will be replaced by the (2 byte) precompile contract address
+  * `0167`: HTS ([HIP-218](https://hips.hedera.com/hip/hip-218))
+  * `016a`: HAS ([HIP-906](https://hips.hedera.com/hip/hip-906))
+  * `016b`: HSS ([HIP-755](https://hips.hedera.com/hip/hip-755))
+* `SSSSSSSS` will be replaced by the (4 byte) redirect method selector
+  * `618dc65e`: `redirectForToken`
+  * `e4cbd3a7`: `redirectForAddress`
+  * `5c3889ca`: `redirectForScheduleTxn`
+* `fefefefefefefefefefefefefefefefefefefefe` will be replaced by the (20 byte) EVM address of the token
+  being redirected, i.e., the HTS token kind's account-num alias (aka: long-zero).
+
+(THe templating in the source code isn't exactly organized this way, but this is the effect.)
+
+## Solidity
+
+Via `library.dedaub.com/decompile`:
+
+```solidity
+function __function_selector__() public nonPayable {
+    CALLDATACOPY(32, 0, msg.data.length);
+    v0 = 0x167.delegatecall(0x618dc65efefefefefefefefefefefefefefefefefefefefe).gas(msg.gas);
+    require(v0 != 0, 0, RETURNDATASIZE()); // checks call status, propagates error data on error
+    return MEM[0:RETURNDATASIZE()];
+}
+```
+
+## How it works
+
+(Example is for HAS system contract, but it works _similarly_ for HTS and HSS.  Not _exactly_
+the same _flow_ for technical reasons, but very similarly.)
+
+1. `FrameBuilder.buildInitialFrameWith` - build a `MessageFrame for this transaction, if it is
+  a contract _call_ then ....
```

Comment:
did you want this ellipses ? and the one below?

Reply:
"style" issue when writing, fixed I think